spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java

```diff
@@ -99,4 +103,32 @@ public static void validatePartitionTransforms(PartitionSpec spec) {
       }
     }
   }
+
+  /**
+   * Pulls any Catalog specific overrides for the Hadoop conf from the current SparkSession, which can be
+   * set via spark.sql.catalog.$catalogName.hadoop.*
+   *
+   * The SparkCatalog allows for hadoop configurations to be overridden per catalog, by setting
+   * them on the SQLConf, where the following will add the property "fs.default.name" with value
+   * "hdfs://hanksnamenode:8020" to the catalog's hadoop configuration.
+   *   SparkSession.builder()
+   *     .config(s"spark.sql.catalog.$catalogName.hadoop.fs.default.name", "hdfs://hanksnamenode:8020")
+   *     .getOrCreate()
+   * @param spark The current Spark session
+   * @param catalogName Name of the catalog to find overrides for.
+   * @return the Hadoop Configuration that should be used for this catalog, with catalog specific overrides applied.
+   */
+  public static Configuration hadoopConfCatalogOverrides(SparkSession spark, String catalogName) {
+    // Find keys for the catalog intended to be hadoop configurations
+    final String hadoopConfCatalogPrefix = String.format("%s.%s.%s", ICEBERG_CATALOG_PREFIX, catalogName, "hadoop.");
+    Configuration conf = spark.sessionState().newHadoopConf();
+    spark.sqlContext().conf().settings().forEach((k, v) -> {
+      // These checks are copied from `spark.sessionState().newHadoopConfWithOptions()`, which we
```

Comment:
I see we lost the synchronize { } call that we would have get from spark.sessionState.newHadoopConf (via getAllConfs method), should we use it as the SynchronizedMap expectes everyone synchronize when getting the map?  (Esp as this is a public API)

Reply:
I was not aware that we lost a synchronize call. I will investigate that and add that in.

Reply:
So I looked into this further, and you're right that `settings` is an instance of `java.util.collections.SynchronizedMap`.

But the `forEach` method of `SynchronizedMap` already has a `synchronized` block of it in the definition.

I tried it with `synchronized` around it and the unit tests passed still, but I'm wondering if it's necessary given that it synchronizes within the class definition itself on the same object (`this` from the class definition).

I'm going to look into this further and then follow up with you offline about this.

Reply:
I added the `synchronized` call as you mentioned, but I'm not sure it's necessary to be honest. If you see the note on the new code, the `forEach` method already uses `synchronize` on itself.

Reply:
As mentioned in another thread, @szehon-ho and I looked into it and the call to `synchronize` is not needed in this place as the underlying map does it on its own (it just doesn't do it when used in scala and calling `.toScala` on it).